pytests/conftest.py

```diff
@@ -280,4 +274,27 @@ def _run(self, cmd, retvalonly=False, cwd=None):
 @pytest.fixture(scope='session')
 def utils():
     test_utils = TestUtils()
+    server = Process(target=TestRepoServer,
+                     args=(test_utils.config['repo_path'], ))
+    server.start()
+
+    atexit.register(StopTestRepoServer, server)
+
+    sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    retry = 0
+    while retry < 10:
+        retry += 1
```

Comment:
Nit: Consider using a `for` loop: `for retry in range(1, 10):`. Just be careful testing the value in line 296.